### PR TITLE
FOUNDATIONS: Azure Storage GetDownloadLink

### PR DESCRIPTION
**Pull request by MichaelOSullivanAnswer**
CLOSES #AB16469

### CODE REVIEW

File: ISL.Providers.Storages.AzureBlobStorage/Brokers/Storages/Blobs/IBlobStorageBroker.cs [hunk 1]
```diff
@@ -5,18 +5,23 @@
...
-        BlobServiceClient blobServiceClient { get; }
+        BlobServiceClient BlobServiceClient { get; }
 
         Response<UserDelegationKey> GetUserDelegationKey(
             DateTimeOffset? startsOn,
             DateTimeOffset expiresOn,
             CancellationToken cancellationToken = default(CancellationToken));
+
+        BlobSasBuilder GetBlobSasBuilder(string blobName, string blobContainerName, DateTimeOffset expiresOn);
+
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Brokers/Storages/Blobs/IBlobStorageBroker.cs:24]: Remove empty line
```suggestion

```

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.Validations.cs [hunk 2]
```diff
@@ -40,12 +40,26 @@ private async ValueTask ValidateStorageArgumentsOnListAsync(string container)
                 (Rule: await IsInvalidAsync(container), Parameter: "Container"));
         }
 
+        private async ValueTask ValidateStorageArgumentsOnGetDownloadLinkAsync(string fileName, string container, DateTimeOffset expiresOn)
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.Validations.cs:43]: wrap long line to be under 120 chars

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs [hunk 3]
```diff
@@ -28,7 +28,7 @@ public ValueTask CreateFileAsync(Stream input, string fileName, string container
                 await ValidateStorageArgumentsOnCreateAsync(input, fileName, container);
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs:28]: ValidateStorageArgumentsOnCreateAsync(input, fileName, container);
```suggestion
                ValidateStorageArgumentsOnCreate(input, fileName, container);
```

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs [hunk 4]
```diff
@@ -41,7 +41,7 @@ public ValueTask RetrieveFileAsync(Stream output, string fileName, string contai
                 await ValidateStorageArgumentsOnRetrieveAsync(output, fileName, container);
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs:41]:
```suggestion
                ValidateStorageArgumentsOnRetrieve(output, fileName, container);
```

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs [hunk 5]
```diff
@@ -54,7 +54,7 @@ public ValueTask DeleteFileAsync(string fileName, string container) =>
                 await ValidateStorageArgumentsOnDeleteAsync(fileName, container);
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs:54]:
```suggestion
                ValidateStorageArgumentsOnDelete(fileName, container);
```

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs [hunk 6]
```diff
@@ -83,7 +83,21 @@ public ValueTask<List<string>> ListFilesInContainerAsync(string container) =>
 
 
         public ValueTask<string> GetDownloadLinkAsync(string fileName, string container, DateTimeOffset expiresOn) =>
-            throw new NotImplementedException();
+            TryCatch(async () =>
+            {
+                await ValidateStorageArgumentsOnGetDownloadLinkAsync(fileName, container, expiresOn);
```
Comment (cjdutoit) [inline ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs:88]:
```suggestion
                ValidateStorageArgumentsOnGetDownloadLink(fileName, container, expiresOn);
```

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs (hunk 5 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs]: this has been updated

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs (hunk 4 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs]: this has been updated

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs (hunk 3 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs]: this has been updated

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.Validations.cs (hunk 2 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.Validations.cs]: this has been updated

File: ISL.Providers.Storages.AzureBlobStorage/Brokers/Storages/Blobs/IBlobStorageBroker.cs (hunk 1 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Brokers/Storages/Blobs/IBlobStorageBroker.cs]: this has been updated

File: ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs (hunk 6 above)
Comment (MichaelOSullivanAnswer) [reply ISL.Providers.Storages.AzureBlobStorage/Services/Foundations/Files/StorageService.cs]: this has been updated

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**